File: frontend/src/data/news.ts
export type NewsStatus = 'PUBLISHED' | 'DRAFT'
export type NewsItem = { id: number; title: string; summary: string; date: string; status: NewsStatus }

export const demoNews: NewsItem[] = [
  { id: 1, title: 'Заседание Координационного совета по взаимодействию с Министерством образования', summary: 'Итоги заседания и ключевые направления совместной работы.', date: '25.06.2026', status: 'PUBLISHED' },
  { id: 2, title: 'Совещание руководителей православных школ и гимназий Московской области', summary: 'Встреча представителей образовательных организаций региона.', date: '02.04.2026', status: 'PUBLISHED' },
  { id: 3, title: 'Материалы для августовского педагогического совета', summary: 'Подборка методических материалов для обсуждения.', date: 'Сегодня, 12:40', status: 'DRAFT' },
]

File: src/main/java/ru/obr_mosmit/site/news/News.java
package ru.obr_mosmit.site.news;

import jakarta.persistence.*;
import java.time.Instant;

@Entity
@Table(name = "news")
public class News {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 300)
    private String title;

    @Column(nullable = false, unique = true, length = 350)
    private String slug;

    @Column(length = 1000)
    private String summary;

    @Column(nullable = false, columnDefinition = "text")
    private String content;

    @Column(name = "cover_image_url", length = 1000)
    private String coverImageUrl;

    @Column(name = "source_url", unique = true, length = 1000)
    private String sourceUrl;
    @Column(name="gallery_urls",columnDefinition="text") private String galleryUrls;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 30)
    private NewsStatus status = NewsStatus.DRAFT;

    @Column(name = "published_at")
    private Instant publishedAt;

    @Column(name = "created_at", nullable = false)
    private Instant createdAt;

    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    @PrePersist
    void onCreate() {
        createdAt = Instant.now();
        updatedAt = createdAt;
    }

    @PreUpdate
    void onUpdate() { updatedAt = Instant.now(); }

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }
    public String getSlug() { return slug; }
    public void setSlug(String slug) { this.slug = slug; }
    public String getSummary() { return summary; }
    public void setSummary(String summary) { this.summary = summary; }
    public String getContent() { return content; }
    public void setContent(String content) { this.content = content; }
    public String getCoverImageUrl() { return coverImageUrl; }
    public void setCoverImageUrl(String coverImageUrl) { this.coverImageUrl = coverImageUrl; }
    public String getSourceUrl() { return sourceUrl; }
    public void setSourceUrl(String sourceUrl) { this.sourceUrl = sourceUrl; }
    public NewsStatus getStatus() { return status; }
    public void setStatus(NewsStatus status) { this.status = status; }
    public Instant getPublishedAt() { return publishedAt; }
    public void setPublishedAt(Instant publishedAt) { this.publishedAt = publishedAt; }
    public Instant getCreatedAt() { return createdAt; }
    public Instant getUpdatedAt() { return updatedAt; }
    public String getGalleryUrls(){return galleryUrls;} public void setGalleryUrls(String value){galleryUrls=value;}
}
